connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -401,6 +402,45 @@ public void testTimestampPropagation() throws Exception {
         PowerMock.verifyAll();
     }
 
+    public void testDisableOffsetCommits() throws Exception {
+        expectInitializeTaskDisableOffsetCommit();
+        expectPollInitialAssignment();
+
+        final List<TopicPartition> partitions = asList(TOPIC_PARTITION, TOPIC_PARTITION2);
+
+        EasyMock.expect(consumer.position(TOPIC_PARTITION)).andReturn(FIRST_OFFSET);
+        EasyMock.expect(consumer.position(TOPIC_PARTITION2)).andReturn(FIRST_OFFSET);
+
+        sinkTask.open(partitions);
+        EasyMock.expectLastCall();
+
+        EasyMock.expect(consumer.poll(EasyMock.anyLong())).andAnswer(
+            new IAnswer<ConsumerRecords<byte[], byte[]>>() {
+                @Override
+                public ConsumerRecords<byte[], byte[]> answer() throws Throwable {
+                    rebalanceListener.getValue().onPartitionsRevoked(partitions);
+                    rebalanceListener.getValue().onPartitionsAssigned(partitions);
+                    return ConsumerRecords.empty();
+                }
+            });
+
+        sinkTask.put(Collections.<SinkRecord>emptyList());
+        EasyMock.expectLastCall();
+
+        consumer.commitAsync(EasyMock.<Map<TopicPartition, OffsetAndMetadata>>anyObject(), EasyMock.<OffsetCommitCallback>anyObject());
```

Comment:
I'm not sure I understand this part of the test. Shouldn't the mock not have any calls to `commitAsync`? I don't think anything in the `WorkerSinkTask` code would cause it to throw an exception. Rather, we expect that *there are no calls* to it. The `verifyAll()` should cause the test to fail if there was an unexpected call to `commitAsync` (or any other method).